### PR TITLE
Re-style the Link Table modal

### LINKED ISSUE #1818: Implement styling and UX improvements for link table modal
- Blocked by https://github.com/centerofci/mathesar/issues/1817

### PR DESCRIPTION
**Pull request by seancolsen**
Fixes #1818

## Notes

https://user-images.githubusercontent.com/42411/207491904-7ab7a487-c48f-4ed7-a652-f2c77a7df42c.mp4

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
@kgodey I've assigned this to you because I'd like you to look over the design comments in my video and request changes if necessary.

**Comment by seancolsen** [comment]
@kgodey 

- (1) to be addressed in #2086 

- (2) addressed in fc2163d7a40e4df376d6878eb8c8297eef78be39

- (3) to be addressed in #2085

- (4) to be addressed in #2084

- (5) addressed in 471849098f5dd90f9429bd7a108fb91bca293de8

**Review by rajatvijay — APPROVED** [review]
Overall LGTM. 

nit: one minor question

File: mathesar_ui/src/components/form/Field.svelte [hunk 1]
```diff
@@ -29,25 +29,27 @@
   $: ({ showsError } = field);
 </script>
 
-<span class="field">
-  <FieldWrapper {label} {layout} {help}>
-    <slot name="help" slot="help" />
+<FieldLayout>
+  {#if label || $$slots.label}
+    <LabeledInput {label} {layout} {help}>
+      <slot name="label" slot="label" />
```
[the remark this answers is not in the document]
Comment (seancolsen) [reply mathesar_ui/src/components/form/Field.svelte:35]: That's an intentional design choice I've used in a few places. The idea here is...

- if you have a plain text label, you can use the prop, which keeps the code more concise in the consuming component
- if you have a rich text label, you can use the slot, which requires slightly more code in the consuming component

I'd be open to re-evaluating this pattern if needed.

**Pull request merged**